### PR TITLE
Add handlers to ensure the child process dies

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Attempt to fix https://github.com/srobo/competition-simulator/issues/114

Webots uses SIGTERM on *nix, and SIGKILL on windows. Webots handling code: https://github.com/cyberbotics/webots/blob/56bab11b4491c03fd6e9dee8fc64f2ee60b6a701/src/webots/control/WbController.cpp#L157

### CODE REVIEW

File: controllers/sr_controller/sr_controller.py [hunk 1]
```diff
@@ -134,14 +135,24 @@ def main():
...
 
-    completed_process = subprocess.run(
+    child_process = subprocess.Popen(
         [sys.executable, "-u", str(robot_file)],
         env=env,
         cwd=str(robot_file.parent),
     )
 
+    def terminate_handler(*args, **kwargs):
+        child_process.terminate()
```
Comment (PeterJCLaw) [inline controllers/sr_controller/sr_controller.py:145]: Should we try to pass on the same signal as we got?

**Comment by antoinepetty** [comment]
It's not working on Windows unfortunately
image

File: controllers/sr_controller/sr_controller.py [hunk 2]
```diff
@@ -134,14 +135,24 @@ def main():
...
+        sys.exit(0)
+
+    # Ensure the child dies
+    signal.signal(signal.SIGKILL, terminate_handler)
+    signal.signal(signal.SIGTERM, terminate_handler)
```
Comment (antoinepetty) [inline controllers/sr_controller/sr_controller.py:150]: This line doesn't seem to work on Windows. However, after removing it the processes still seem to stick around when resetting webots.

Comment (trickeydan) [reply controllers/sr_controller/sr_controller.py:150]: Might be worth looking into [PSUtil](https://pypi.org/project/psutil/) for this.

If we don't want to add another dependency, it is BSD licensed, so we can just select the right parts of that library to include in our code.

**Comment by PeterJCLaw** [comment]
Replaced by #150.
Thanks for investigating this approach @RealOrangeOne.

**Pull request closed without merging**